cmd/neofs-cli/modules/request/container.go

```diff
@@ -0,0 +1,239 @@
+	defer cancel()
+	// FIXME: inlined from https://github.com/nspcc-dev/neofs-sdk-go/blob/e9a774f505257dc48f005073162713805bdefebc/client/client.go#L178
+	//  use new approach
+	//nolint:staticcheck
+	conn, err := grpc.DialContext(dialCtx, addr,
+		grpc.WithTransportCredentials(transportCreds),
+		grpc.WithReturnConnectionError(),
+		grpc.FailOnNonTempDialError(true),
+	)
+	if err != nil {
+		return fmt.Errorf("dial endpoint URI with gRPC: %w", err)
+	}
+	defer conn.Close()
+
+	cmd.Println("Connected. Sending request...")
+
+	reqCtx, cancel := context.WithTimeout(cmd.Context(), defaultRequestTimeout)
+	defer cancel()
+	cli := protocontainer.NewContainerServiceClient(conn)
+	resp, err := cli.Put(reqCtx, req)
+	if err != nil {
+		return fmt.Errorf("transport failure: %w", err)
+	}
+
+	cmd.Println("Response received. Checking signatures...")
+
+	if err := neofscrypto.VerifyResponseWithBuffer(resp, nil); err != nil {
+		return fmt.Errorf("failed to verify response signatures: %w", err)
+	}
+
+	cmd.Println("Signatures are valid. Checking status...")
+
+	if err := apistatus.ToError(resp.GetMetaHeader().GetStatus()); err != nil {
+		return fmt.Errorf("status failure: %w", err)
+	}
+
+	cmd.Println("Status OK. Checking body...")
+
+	if resp.Body == nil {
+		return errors.New("missing response body")
+	}
+	if resp.Body.ContainerId == nil {
+		return errors.New("invalid response body: missing container ID field")
+	}
+	var id cid.ID
+	if err := id.FromProtoMessage(resp.Body.ContainerId); err != nil {
+		return fmt.Errorf("invalid response body: invalid container ID field: %w", err)
+	}
+
+	defer cmd.Printf("Container ID: %s\n", id)
+
+	cmd.Println("Request accepted for processing. Polling success...")
+
+	getReq := &protocontainer.GetRequest{
+		Body: &protocontainer.GetRequest_Body{
+			ContainerId: resp.Body.ContainerId,
+		},
+	}
+	if getReq.VerifyHeader, err = neofscrypto.SignRequestWithBuffer((*neofsecdsa.Signer)(pk), getReq, nil); err != nil {
+		return fmt.Errorf("failed to sign Get request with generated ECDSA private key: %w", err)
+	}
+
+	pollStart := time.Now()
+	for attempt := 1; ; attempt++ {
+		cmd.Println("Requesting container...")
+
+		reqCtx, cancel := context.WithTimeout(cmd.Context(), defaultRequestTimeout)
+		resp, err := cli.Get(reqCtx, getReq)
+		cancel()
+		if err != nil {
+			return fmt.Errorf("transport failure: %w", err)
+		}
+
+		cmd.Println("Response received. Checking signatures...")
+
+		if err := neofscrypto.VerifyResponseWithBuffer(resp, nil); err != nil {
+			return fmt.Errorf("failed to verify response signatures: %w", err)
+		}
+
+		cmd.Println("Signatures are valid. Checking status...")
+
+		if err := apistatus.ToError(resp.GetMetaHeader().GetStatus()); err == nil {
+			cmd.Println("Status OK. Operation succeeded.")
+			return nil
+		} else if !errors.Is(err, apistatus.ErrContainerNotFound) {
+			return fmt.Errorf("status failure: %w", err)
+		}
+
+		const maxAttempts = 3
```

Comment:
try based on user inputs? like timeout in our every command?

Reply:
it's 10s now. In practice, container is either created fast, or never

i tend to think that good tools work without explicitly specified timeout in 99% cases cuz timeout values they are often too opaque